contracts/src/instances/ezeth/EzETHBase.sol

```diff
@@ -0,0 +1,146 @@
+        // Transfer ezETH shares into the contract.
+        _ezETH.transferFrom(msg.sender, address(this), _shareAmount);
+    }
+
+    /// @dev Process a withdrawal in base and send the proceeds to the
+    ///      destination.
+    function _withdrawWithBase(
+        uint256, // unused
+        address, // unused
+        bytes calldata // unused
+    ) internal pure override returns (uint256) {
+        // ezETH withdrawals aren't necessarily instantaneous. Users that want
+        // to withdraw can manage their withdrawal separately.
+        revert IHyperdrive.UnsupportedToken();
+    }
+
+    /// @dev Process a withdrawal in vault shares and send the proceeds to the
+    ///      destination.
+    /// @param _shareAmount The amount of vault shares to withdraw.
+    /// @param _destination The destination of the withdrawal.
+    function _withdrawWithShares(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override {
+        // Transfer the ezETH shares to the destination.
+        _ezETH.transfer(_destination, _shareAmount);
+    }
+
+    /// @dev Convert an amount of vault shares to an amount of base.
+    /// @param _shareAmount The vault shares amount.
+    /// @return baseAmount The base amount.
+    function _convertToBase(
+        uint256 _shareAmount
+    ) internal view override returns (uint256) {
+        // Get the total TVL priced in ETH from restakeManager
+        (, , uint256 totalTVL) = _restakeManager.calculateTVLs();
+
+        // Get the total supply of the ezETH token
+        uint256 totalSupply = _ezETH.totalSupply();
+
+        return
+            _renzoOracle.calculateRedeemAmount(
+                _shareAmount,
+                totalSupply,
+                totalTVL
+            );
+    }
+
+    /// @dev Convert an amount of base to an amount of vault shares.
+    /// @param _baseAmount The base amount.
+    /// @return shareAmount The vault shares amount.
+    function _convertToShares(
+        uint256 _baseAmount
+    ) internal view override returns (uint256) {
+        // Get the total TVL priced in ETH from restakeManager
+        (, , uint256 totalTVL) = _restakeManager.calculateTVLs();
+
+        // Get the total supply of the ezETH token
+        uint256 totalSupply = _ezETH.totalSupply();
+
+        return
+            _renzoOracle.calculateMintAmount(
+                totalTVL,
+                _baseAmount,
+                totalSupply
+            );
+    }
+
+    /// @dev Unused, gets the total amount of base held by the pool.
+    /// @return baseAmount The total amount of base.
+    function _totalBase() internal pure override returns (uint256) {}
+
+    /// @dev Unused, gets the total amount of shares held by the pool in the yield
+    ///      source.
+    /// @return shareAmount The total amount of shares.
+    function _totalShares()
+        internal
+        view
+        override
+        returns (uint256 shareAmount)
+    {}
```

Comment:
Why are these unused? We need both of these to be implemented. Refer to Cash's PRs.

Reply:
Sorry I just changed these because I thought they weren't used, just tracked down that they are used for `sweep()`

Reply:
Put these back in.